pkg/plugins/resourcestream/http_response_streamer.go

```diff
@@ -173,3 +173,44 @@ func (s *callResourceResponseStream) Close() error {
 	s.closed = true
 	return nil
 }
+
+type HTTPResponseSender struct {
```

Comment:
Started to questioning why we actually need to do the streaming in a new goroutine. Tried this instead and seems to work. Handling an HTTP request is already in its own goroutine and the call to `CallResource` will not return until the sender has sent all messages.

@grafana/plugins-platform-backend  do you see any problems with this?approach?

Reply:
Thanks, yes that's what I'm trying to figure out: 

As per the Go HTTP server [Serve description](https://github.com/golang/go/blob/39957b5d89fec8bc3a79f4a982452c6e3d5b3ad3/src/net/http/server.go#L3011-L3021) it should be the case:
> Serve accepts incoming connections on the Listener l, creating a
new service goroutine for each. The service goroutines read requests and
then call srv.Handler to reply to them.

Given HTTP request handler is called in its own goroutine and given
- gRPC plugin it will not return until it has consumed the stream:  https://github.com/grafana/grafana/blob/4b3aead2d053d655a3730ad855ff2141229f9cf3/pkg/plugins/backendplugin/grpcplugin/client_v2.go#L186-L203
- core plugin it will just delegate the sending/handling part to a core plugin:
https://github.com/grafana/grafana/blob/4b3aead2d053d655a3730ad855ff2141229f9cf3/pkg/plugins/backendplugin/coreplugin/core_plugin.go#L92-L98

My conclusion is this should work so I'm going to make needed changes.